### PR TITLE
otto: add 0.1.1 (new formula)

### PR DESCRIPTION
**Pull request by chrisfu**
Knocked up a quick formula to build Otto, which was recently announced by HashiCorp as a replacement for Vagrant. Based this on the pre-existing formula for Vault. Audited and confirmed working in Yosimite.

Thanks.

### CODE REVIEW

**Comment by chrisfu** [comment]
Looks as if this is an effective pass now. The free Github API limit caused the CI fail.

File: Library/Formula/otto.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  end
+
+  # gox dependency
+  go_resource "github.com/mitchellh/iochan" do
+    url "https://github.com/mitchellh/iochan.git",
+        :revision => "87b45ffd0e9581375c491fef3d32130bb15c5bd7"
+  end
+
+  def install
+    contents = Dir["{*,.git,.gitignore,.travis.yml}"]
```
Comment (bfontaine) [inline Library/Formula/otto.rb]: Why do we need the `.travis.yml` file?

**Comment by chrisfu** [comment]
@bfontaine Hmm, are these legitimate failures on my part, or is this indeed effectively a pass?

**Comment by bfontaine** [comment]
@chrisfu No it’s an issue on the bot; I’ll re-trigger it.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

File: Library/Formula/otto.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    end
+
+    cd gopath/"src/github.com/hashicorp/otto" do
+      system "make", "dev"
+      bin.install "bin/otto"
+    end
+  end
+
+  test do
+    system "#{bin}/otto", "--version"
```
Comment (dunn) [inline Library/Formula/otto.rb]: Any chance of a test that checks the functionality a bit more?

**Comment by chrisfu** [comment]
@dunn I'd imagine so. However, as it's a new tool that does things quite differently to Vagrant, I (and anyone else interested in this) would need to have a think about how best to test it without spawning an entire development environment. :)

It's something that I'm giving some though to, but I don't think the fairly weak test should be any sort of barrier to merging for now.

@bfontaine Any change of triggering another build please? A build failing/passing seems like roulette sometimes, heh.

**Comment by dunn** [comment]
Don't worry about the Travis build failing, as long as Jenkins passes.

**Comment by AndrewDryga** [comment]
+1

**Comment by dunn** [comment]
Looking closer at the build process, it uses `go get` (https://github.com/hashicorp/otto/blob/master/scripts/build.sh#L24) without an accompanying `Godeps.json`.  That means it's just pulling the HEAD of each dependency, right?

We try to avoid that, which is why we have the `go_resource` method.  I know it's a hassle, but could the (recursive) dependencies be specified in the formula?  Alternatively you could ask the Otto developers to add a `Godeps.json` with dependency revisions.

**Comment by bfontaine** [comment]
@dunn `go get` does pull the HEAD of each dependency; but the `Godeps.json` file is related to `godep`, a Go vendoring tool, and not to `go` itself. That means running `go get ./...` with or without a `Godeps.json` file does the same thing.
We could either ask upstream for specific revisions or just use whatever the revision currently is for each dependency and put that in `go_resource` blocks.

**Comment by dunn** [comment]
Oh, gotcha. Is the Godep we're vendoring used by otto at runtime, then, rather than being used to build otto?

**Comment by bfontaine** [comment]
No, `Godep` is only used at build-time; Go uses static linking by default.

**Comment by maxnordlund** [comment]
How is this going?

**Comment by bfontaine** [comment]
We need `go_resource`s for the external dependencies per @dunn’s comment. Right now the `build.sh` fetches external dependencies; we must put them in `go_resource`s.

The following should work:

1. Run `brew fetch otto`
2. Disable your Internet access
3. Run `brew install otto`

Right now it fails:
```
…snip…
==> make dev
find . -type f -name '.DS_Store' -delete
go generate ./...
==> Getting dependencies...
# cd .; git clone https://github.com/hashicorp/go-checkpoint /private/tmp/otto20151021-53973-n08so6/gopath/src/github.com/hashicorp/go-checkpoint
Cloning into '/private/tmp/otto20151021-53973-n08so6/gopath/src/github.com/hashicorp/go-checkpoint'...
fatal: unable to access 'https://github.com/hashicorp/go-checkpoint/': Could not resolve host: github.com
package github.com/hashicorp/go-checkpoint: exit status 128
…more errors…
```

**Comment by DomT4** [comment]
Closing in favour of #45525, which seems more or less complete. Thanks for the submission nonetheless @chrisfu! :heart:

**Pull request closed without merging**